mlflow/dspy/callback.py

```diff
@@ -111,15 +115,15 @@ def _start_span(
         attributes: dict[str, Any],
     ):
         # Get parent span in this order:
-        # 1. If there is an parent component in DSPy, use its span as parent span.
-        # 2. If there is an active span in MLflow, use it as parent span.
+        # 1. If there is an active span in MLflow, use it as parent span.
+        # 2. Use DSPy's active call ID to find the parent span as a fallback.
         # 3. Otherwise, start a new root span.
-        if parent_call_id := ACTIVE_CALL_ID.get():
-            parent_span = self._call_id_to_span.get(parent_call_id)
+        if active_span := mlflow.get_current_active_span():
+            parent_span = active_span
+        elif parent_call_id := ACTIVE_CALL_ID.get():
+            parent_span, _ = self._call_id_to_span.get(parent_call_id)
```

Comment:
Is dataclass unpackable?

Reply:
`self._call_id_to_span.get(parent_call_id)` returns a dataclass object, right?

Reply:
You're right. Indeed this will be a dead code once we added the token logic. Let me remove it.